Submission fails with every package specification mechanism

Context: I am trying to release [ocamlformat 0.8](
https://github.com/ocaml-ppx/ocamlformat/archive/0.8.tar.gz).

With opam-publish 2.0.0, all of the ways of specifying a package to submit fail. The errors aren't leading me to the problem (which could well be my fault).

```
$ opam publish ocamlformat.opam

The following will be published:
  - ocamlformat version 0.8 with opam file at /Users/jjb/ocamlformat/ocamlformat.opam

[WARNING] These will be virtual packages: ocamlformat.0.8

Continue ? You will be shown the patch before submitting. [Y/n] y
[ERROR] Uncaught exception: SSL: Method error
[Exit 99]
$ opam publish --tag=0.8 ocaml-ppx/ocamlformat
[ERROR] Uncaught exception: "/bin/cp -PRp
        /var/folders/qd/00gsz33d7zj71k_dhp7bwv5cxm01z7/T/opam-33717-58c514/ocamlformat-0.8
        /private/var/folders/qd/00gsz33d7zj71k_dhp7bwv5cxm01z7/T/opam-33717-5a61c5/sources/\0008\197C\1346\002\138\n\175\022h\2269\170E-0.8.tar.gz"
        exited with code 127
[Exit 99]
$ opam publish https://github.com/ocaml-ppx/ocamlformat/archive/0.8.tar.gz
[ERROR] Uncaught exception: "/bin/cp -PRp
        /var/folders/qd/00gsz33d7zj71k_dhp7bwv5cxm01z7/T/opam-33786-58c514/ocamlformat-0.8
        /private/var/folders/qd/00gsz33d7zj71k_dhp7bwv5cxm01z7/T/opam-33786-5a61c5/sources/\0008\197C\1346\002\138\n\175\022h\2269\170E-0.8.tar.gz"
        exited with code 127
[Exit 99]
```
I assumed that #58 is still causing trouble, and so executed `opam install lwt_ssl; opam reinstall opam-publish`, but it did not help.

Any hints?

I also don't know why there was a message about the package being 'virtual'.

--- a/src/publishMain.ml
+++ b/src/publishMain.ml
@@ -48,7 +48,7 @@ let tmp_archive tmpdir url =
 let tmp_source tmpdir url =
   OpamFilename.Op.(tmpdir / "sources" /
                    Printf.sprintf "%s-%s"
-                     (Digest.string (OpamUrl.to_string url))
+                     (Digest.to_hex (Digest.string (OpamUrl.to_string url)))
                      (OpamUrl.basename url))
 
 let upgrade_to_2_0 ?(local=true) opam0 =

